config/v1/types_oauth.go

```diff
@@ -204,111 +184,92 @@ type IdentityProviderConfig struct {
 
 	// Provider-specific configuration
 	// The json tag MUST match the `Type` specified above, case-insensitively
-	// e.g. For `Type: "LDAP"`, the `LDAPPasswordIdentityProvider` configuration should be provided
+	// e.g. For `Type: "LDAP"`, the `ldap` configuration should be provided
 
 	// basicAuth contains configuration options for the BasicAuth IdP
 	// +optional
-	BasicAuth *BasicAuthPasswordIdentityProvider `json:"basicAuth,omitempty"`
+	BasicAuth *BasicAuthIdentityProvider `json:"basicAuth,omitempty"`
 
-	// allowAll enables the AllowAllIdentityProvider which provides identities for users
-	// authenticating using non-empty passwords.
-	// Defaults to `false`, i.e. allowAll set to off
+	// github enables  user authentication using GitHub credentials
 	// +optional
-	AllowAll bool `json:"allowAll,omitempty"`
```

Comment:
I can't think of a reason we need to keep this.  If anyone can, mark it here.

Reply:
I will drop it.